### PR TITLE
Val/revert66430 17.6

### PR DESCRIPTION
**Pull request by chsienki**
Port of https://github.com/dotnet/roslyn/pull/67808 (a revert PR) to 17.6 for servicing
Corresponding DevDiv issue: https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1733125

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -12,7 +12,9 @@
     <add key="dotnet-public" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public/nuget/v3/index.json" />
     <add key="nuget-build" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/nuget-build/nuget/v3/index.json" />
     <add key="vssdk" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json" />
+    <add key="vssdk-archived" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk-archived/nuget/v3/index.json" />
```
Comment (AlekseyTs) [inline NuGet.config:15]: ><add key="vssdk-archived" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk-archived/nuget/v3/index.json" /> [](http://example.com/codeflow?start=4&length=128)

Changes in this file look unrelated to the revert

**Comment by AlekseyTs** [comment]
Done with review pass (commit 2). It looks like CI jobs are failing, BTW.

File: NuGet.config (hunk 1 above)
Comment (chsienki) [reply NuGet.config:15]: VS has changed the way they handle NuGet feeds since we last inserted this branch. This is a cherry pick from main that is required to build this branch.

**Review by jaredpar — APPROVED** [review]

File: NuGet.config (hunk 1 above)
Comment (AlekseyTs) [reply NuGet.config:15]: > This is a cherry pick from main that is required to build this branch.

The change shouldn't be part of this PR, unless it is specific to the change we are making. Which it sounds like it is not related.

Comment (chsienki) [reply NuGet.config:15]: Ok, I can break it out into two PRs, but they'll be inserted in the same VS insertion either way. We can't insert without the feed change.

**Comment by chsienki** [comment]
Updated to only include the single commit

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1)
